packages/dds/task-manager/src/taskManager.ts

```diff
@@ -337,7 +338,7 @@ export class TaskManagerClass
 
 		// This promise works even if we already have an outstanding volunteer op.
 		const lockAcquireP = new Promise<boolean>((resolve, reject) => {
-			const currentMessageId = this.messageId;
+			const nextPendingMessageId = this.nextPendingMessageId;
```

Comment:
It took me a minute to understand why this is OK even if we don't end up sending a volunteer op - might benefit from a comment indicating it's intentional, maybe like
```suggestion
			// If we don't send an op (meaning the latest pending op is "volunteer"), nextPendingMessageId
			// will be greater than that prior "volunteer" op's messageId.  This is OK because
			// we only use it to filter stale abandon/complete, and not when determining if we
			// acquired the lock.
			const nextPendingMessageId = this.nextPendingMessageId;
```